### ISSUE TITLE
Subsequent citations automatically surrounded by round brackets in note style

### ISSUE THREAD

**Issue by ah-dagdelen** [jgm/pandoc#7569]
Subsequent citations in footnoes are surrounded by round brackets in the output format. The period at the citation end is also ignored. Neither the input text nor the CSL style file have round bracket character in/for the relevant positions. They are added to text in the conversion process.

[r.zip](https://github.com/jgm/citeproc/files/7163704/r.zip)
Enclosed, I upload the relevant files. Citations after the second \cites command are set between brackets in the output format.

System
Pandoc version on Linux Mint 20.2 Cinnamon I use is as follows: Compiled with pandoc-types 1.22, texmath 0.12.3, skylighting 0.11, citeproc 0.4.1, ipynb 0.1.0.1

Command line
pandoc -C r.org --bibliography=r.json --csl=r.csl -o r.docx

**Comment by ah-dagdelen** [comment]
If I'm not mistaken, there is a misunderstanding here.

You don't switch between author-date and note style. You switch between in-text class and note class. Author-date style is citation format related, not class related. You can use the same author-date style with note class just by changing the class type. And then you get the same author-date style for your citations in your footnotes.

If you have the following command in your style file, you should get brackets for your citations whether you use in-text class or note class. Here, the style is not relevant.

```
  <citation>
    <layout prefix="(" suffix=")">
    </layout>
  </citation>
```
If you don't have this command, you won't get brackets in author-date style. If you have this command, you will get brackets for your citations in footnotes.

If a citation is already in a footnote, then it should be printed just like the other citations in the same footnote. Please refer to my example file for this issue.

If a citation is already in a footnote, then the converter should not create a new footnote for that citation. This is of course here not the case. So, this is not the problem here we are referring. I wrote this just to make the problem clear.

Thanks again for your help and for your time.

**Comment by jgm** [comment]
I don't think you're understanding. Let me try again.

> You don't switch between author-date and note style. You switch between in-text class and note class

Not sure what you mean here (is this class terminology org-specific?), but CSL styles are classed as either author-date, numerical, or note. The style file will contain
```
    <category citation-format="note"/>
```
That's the distinction I'm referring to.

You're right, CSL just gives you a formatted citation, and in note styles, this won't contain parentheses typically.  But pandoc then has to do the work of integrating this citation into the surrounding text.  If the citation occurs in the main text and it's a note style, pandoc does this by creating a note with the (unmodified) citation content.  To be explicit,
```
This [@smith2000].
```
gets rendered as
```
This[^1].

[^1]. John Smith, Title, 2000, etc.
```
This is great, because you can go back and forth between note and author-date styles just by using a different CSL style!

The problem is what to do about citations that occur inside manually added footnotes, like
```
[^1]:  Similar numbers have been recorded in Syria [@jones2000, p. 20] and in Iraq [@smith2000, p. 33].
```
If we did what we normally do, these would be replaced by footnotes, but we don't want footnotes inside footnotes. So instead, pandoc puts them in parentheses.  If we didn't do that, we'd get ungrammatical gibberish:
```
[^1]:  Similar numbers have been recorded in Syria Jim Jones, Title, 2000, p. 20 and in Iraq Sam Smith, Title, 2000, p. 33.
```
Anyway, I hope that explains why it works the way it does. One could discuss whether setting these off with commas instead of parentheses would be a better choice, but something is needed here.

**Comment by badumont** [comment]
If I can permit myself, I think that the problem should not arise using citeproc in the the classic way, i.e. with a markdown file, since the manual clearly states : 

  > When you’re using a note style, avoid inserting footnotes manually. Instead, insert citations just as you would in an author-date style.

The problem arises from the fact that, in the Org file, cites occur inside footnotes. Maybe the problem could be resolved by enabling the reader to turn those entire footnotes into Citation objects if a metadata field such as `csl-class` is set to note?

**Comment by ah-dagdelen** [comment]
We should be referring to CSL class not to category. Every CSL file has to define its class at the very beginning. So, a style file has to start like this:

```
<?xml version="1.0" encoding="utf-8"?>
<style xmlns="http://purl.org/net/xbiblio/csl" version="1.0" class="note" default-locale="en-US">
```
As we can see, there is a class information there. Without that, we cannot have the CSL file work. We have there two options: in-text or note. CSL styles are classed as in-text or note. If we choose note class, our citations are rendered as footnotes or in footnotes. So, when I make a new citation in a footnote, I don't get a new footnote for that citation, I don't get a footnote inside a footnote. But when I make a new citation in text, then I get a new footnote for that citation. This is how CSL style works.

What you are referring to is category names or style names. CSL category has nothing to do with the citation formatting/rendering. It is set in order to make clear what type of style we get, like a short info about the style. It helps cataloging styles if I am not mistaken, it does nothing more. Furthermore, it is optional. As you can see, there is no category information in the enclosed style file. But CSL file still works.

Pandoc or citeproc should let the CSL style file to decide whether a citation will be placed in brackets/parentheses or not. There is a layout option for that. And it has nothing to do with CSL category or class. We define the formatting under citation layout or in macros. In note class, we get parentheses when we tell CSL file to do so. We don't get parentheses, notwithstanding CSL category or class, when we leave suffix and prefix field empty.

There is no correlation between putting citations in brackets/parentheses and the style category or class.

As to second point, it should be up to author's decision whether the citations in footnotes are rendered in parentheses or not.

`[^1]:  Similar numbers have been recorded in Syria [@jones2000, p. 20] and in Iraq [@smith2000, p. 33].`

In this example, one can manually put the citation key in parentheses in the input file if necessary. If we do what we normally should do, these should not be replaced by footnotes or create footnotes in footnotes, as they are already in footnotes. These should be rendered in accordance with the CSL style file in use. There will be no ungrammatical gibberish by default. It is up to author's decision whether he/she puts citations in parentheses or not. If there is nothing surrounding, we cannot simply add something in the conversion process. It is not up to Pandoc's decision, it is up to author's decision how the citations are printed.

Furhermore, it is also possible to edit CSL style file to get citations occurred inside manually added footnotes in parentheses. So, there is no need Pandoc does this automatically.

If my explanation sounds rude I beg you pardon. I cannot notice small details, which can make the text unpleasant for native speakers.

**Comment by jgm** [comment]
> As we can see, there is a class information there. Without that, we cannot have the CSL file work. We have there two options: in-text or note. CSL styles are classed as in-text or note. If we choose note class, our citations are rendered as footnotes or in footnotes. So, when I make a new citation in a footnote, I don't get a new footnote for that citation, I don't get a footnote inside a footnote. But when I make a new citation in text, then I get a new footnote for that citation. This is how CSL style works.

CSL processors are responsible for generating a formatted citation.  But, as noted above, pandoc then has to figure out how to integrate this citation into the document.  I've explained above the choices we've made there -- motivated by the desire to switch easily from an in-text style to a note style -- and why the problems arise.  Please look back at this part of what I said.

> In this example, one can manually put the citation key in parentheses in the input file if necessary.

You mean one could write:

```
[^1]: Similar numbers have been recorded in Syria ([@jones2000, p. 20]) and in Iraq ([@smith2000, p. 33]).
```
Yes, if pandoc just inserted the raw formatted citation, users could still do this. But that would break the design goal I mentioned above.  If you switched to an author-date style, you'd then get doubled parentheses, because the CSL processor would already have inserted its own set of parentheses.

**Comment by ah-dagdelen** [comment]
This design goal has no meaning for someone not willing to switch between styles. It produces a wrong output. Most of the time, we know at the beginning what type of style we need.

We can omit double parentheses around citations in the conversion process.

**Comment by ah-dagdelen** [comment]
And I suspect whether there is a parentheses problem on the basis of the given example which should be solved by this design goal when we switch between in-text and note styles.

If your style file has parentheses for your citations, then you will get parentheses for your citations. Switching between in-text and note style has nothing to do with this. So, there is no need to think about how we switch easily from an in-text to a note style in this regard.

**Comment by jgm** [comment]
> This design goal has no meaning for someone not willing to switch between styles

Pandoc is designed to have a certain feature. You don't care about that feature. I understand that.  But many people do use this feature; I'm just trying to explain why it behaves the way it does.

I'm happy to consider alternatives, as long as they're consistent with this design goal.  The alternative of just inserting the citation with no surrounding punctuation is not consistent with it, because it means that many citations that work fine with in-text styles will produce ungrammatical gibberish when formatted with a note style.

I don't know much about how citations are handled in org (and I understand this is quickly evolving). But in pandoc's markdown, we have adopted a way of getting around this difficulty: author-in-text form citations (`@cite` as opposed to `[@cite]`) are inserted into notes without parentheses (though sometimes a comma is added).  Thus:
```
[^1]:  Here's my manually added note @cite1.  
```
will render as
```
1. Here's my manually added note, John Smith, Title, etc.
```
This is documented in the manual.  So if you change your source to
```
[fn:1] \citet[s. 18]{atamerUluslararasiSatimSozlesmelerine2005} \citet[s. 17]{serozanBorclarHukukuOzel2019} some text \citet[s. 17]{serozanBorclarHukukuGenel2016} \citet[s. 16]{atamerGenelIslemKosulu2013}
```
you'll get this:
```
[1] Yeşim Atamer, Uluslararası satım sözleşmelerine ilişkin Birleşmiş
Milletler Antlaşması (CISG) uyarınca satıcının yükümlülükleri ve
sözleşmeye aykırılığın sonuçları, İstanbul, Beta, 2005, s. 18 Rona
Serozan/Başak Baysal/Kerem Cem Sanlı, Borçlar Hukuku Özel Bölüm, 4. bs.,
İstanbul, On İki Levha, 2019, s. 17 some text Rona Serozan, Borçlar
Hukuku Genel Bölüm üçüncü cilt: İfa, ifa engelleri, haksız zenginleşme,
ed. Necip Kocayusufpaşaoğlu/Hüseyin Hatemi/Abdülkadir Arpacı, 7. bs.,
İstanbul, Filiz Kitabevi, 2016, s. 17 Yeşim Atamer, “Genel işlem koşulu
mu bireysel pazarlıkla kurulan sözleşme mi?”, Yeni Türk Borçlar Kanunu
ve Yeni Türk Ticaret Kanunu Sempozyumu, ed. Çiğdem Kırca, İstanbul,
Vedat Kitapçılık, 2013, ss. 103-137, s. 16.
```

You may ask: why not just have it work this way for ALL citations (not just author-in-text) in notes?  Well, that's covered in the linked discussions and issues.  But basically it's because this will give bad results for many uses of citations in notes that work fine in author-date styles.  (So, the design goal.)

What if you don't want to change your org file?  Well, then you can work around this by using a Lua filter that changes all citations occuring in notes to AuthorInText style.  To start you out, here's one that just changes all Cite objects so that they are author-in-text (regardless of whether they're in notes -- if you have citations outside of notes, you'll need something more complex):
```lua
function Cite(el)
  el.citations[1].mode = "AuthorInText"
  return el
end
```
Apply this using `--lua-filter` *before* `--citeproc` on the command line.

**Comment by ah-dagdelen** [comment]
I understand this design decision solves a lot of problems, and help people. But on the other side, it also creates some problems, or at least has the potential. My case is one of these. I am the one who conceded this design's goal.

Because I sort my citations, the citations must be covered by the same citation command. I use `\cites` for that. Another command will not work I assume.

As a workaround for my problem, I converted my CSL style file from note class to in-text class. As long as I do not add citations in text, this should work, and won't make an issue. I always add citations in manually created footnotes. This the way I am going to use in order to have Pandoc convert documents.

I tried to explain my point of view on this issue. I am still suspicious about that this design decision is a right idea. However, I always bear in mind that I might be wrong. Maybe, I will see it after some time.

Thank you for the clarifications and solutions you supplied.
